modules/collection/programs/hyprpanel.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, rumLib, ... }: let
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkOption mkEnableOption mkPackageOption;
+  inherit (lib.generators) toJSON;
+  inherit (lib.types) nullOr either str path attrs;
+
+  cfg = config.rum.programs.hyprpanel;
+in {
+  options.rum.programs.hyprpanel = {
+    enable = mkEnableOption "HyprPanel";
+
+    package = mkPackageOption pkgs "hyprpanel" {};
+
+    settings = mkOption {
+      type = attrs;
+      default = {};
+      example = {
+        bar = {
+          autoHide = "never";
+          location = "top";
+        };
+        theme = {
+          name = "catppuccin-mocha";
+          font = "JetBrains Mono Nerd Font";
+        };
+      };
+      description = ''
+        Configuration written to {file}`$HOME/.config/hyprpanel/config.json`.
+        Refer to [HyprPanel documentation](https://hyprpanel.com/configuration).
+      '';
+    };
+
+    config = mkOption {
+      type = nullOr (either path (either str attrs));
+      default = null;
+      description = "Alternative config input (overrides settings if set)";
+    };
```

Comment:
1. Do we need a verbatim `extraConfig` option for `hyprpanel`?
2. This would *append* to anything in `settings`, not override. It's simply for options that either can't be expressed in `settings` or better written verbatim.